modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/portlet/action/GetTrafficSourcesMVCResourceCommand.java

```diff
@@ -160,24 +165,26 @@ protected void doServeResource(
 		}
 
 		try {
-			Map<String, TrafficSource> trafficSources =
-				analyticsReportsDataProvider.getTrafficSources(
+			Map<String, TrafficChannel> trafficChannels =
+				analyticsReportsDataProvider.getTrafficChannels(
 					companyId, canonicalURL);
 
 			emptyMap.forEach(
-				(name, trafficSource) -> trafficSources.merge(
-					name, trafficSource,
-					(trafficSource1, trafficSource2) -> trafficSource1));
+				(name, trafficChannel) -> trafficChannels.merge(
+					name, trafficChannel,
+					(trafficChannel1, trafficChannel2) -> trafficChannel1));
 
-			return new ArrayList<>(trafficSources.values());
+			return new ArrayList<>(trafficChannels.values());
 		}
 		catch (PortalException portalException) {
 			_log.error(portalException, portalException);
 
 			return Arrays.asList(
-				new TrafficSource("direct"), new TrafficSource("organic"),
-				new TrafficSource("paid"), new TrafficSource("referral"),
-				new TrafficSource("social"));
+				DirectTrafficChannel.withError(),
```

Comment:
If we want to use the withError, we should create a builder and call new DirectTrafficChannelBuilder().withError()

Reply:
see comment above, I've added a new constructor ;)